### ISSUE TITLE
Unable to create multiple foreign keys

### ISSUE THREAD

**Issue by sujkid** [ga-wdi-boston/full-stack-project#515]
I'm trying to create 2 foreign keys that reference to the same table, but I keep getting an error while I try to migrate..

rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::DuplicateColumn: ERROR:  column "user_item_id" of relation "requests" already exists

I referenced #334
but I still can't get it to work for me.. 
I think its too late to redesign my tables at this point?

**Comment by neugierige** [comment]
can you go into your database and do `\d [table name]` to see what is currently in the table?

**Comment by gaand** [comment]
You'll need to name one of the foreign keys a different name.

The first result here might help:

https://www.google.com/#q=rails+two+foreign+keys+to+same+table

**Comment by sujkid** [comment]
my foreign key names are different! 
This is how I created the foreign keys..

rails g migration AddReqUserItem1ToRequests user_item:references

rails g migration AddReqUserItem2ToRequests user_item:references
(the names are a little unimaginative)
and i get two migration files

but on bundle exec rake db:migrate
i get the above error

am i doing the above foreign key creation wrong?

**Comment by gaand** [comment]
This creates the name of the column:`user_item:references`.  

So, the names of your migrations are different, but not the names those migrations try to create.

**Comment by sujkid** [comment]
yes, in my table, I actually have the foreign key as user_item_id
thats not what i want though.. 
i was actually trying to create foreign keys by the name
req_user1_item_id and req_user2_item_id
and reference them to my user_items table

**Comment by gaand** [comment]
This may help

http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference

**Comment by sujkid** [comment]
so i realized both my migrations for the foreign keys had the same name, 
so in the add_reference, i changed the names of the keys???
and I get this error:

rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::UndefinedTable: ERROR:  relation "requestsUserItem1" does not exist

**Comment by raq929** [comment]
What's your status?

**Comment by sujkid** [comment]
moving on to other problems for now.. :)

**Comment by raq929** [comment]
So, was this resolved? If yes, please comment. If no, please comment with what you have tried.

**Comment by sujkid** [comment]
I am keeping this open for now, some of the other migrations that I tried, like changing the names and editing the Models failed. I am a little burned out on this issue.. so I will come back to it after I implement another feature in my project
something tells me this should not be hard to do though..

**Comment by neugierige** [comment]
so the issue is that Suja was trying to create 2 foreign keys that refer to the SAME table. 

I think this is what you need: http://guides.rubyonrails.org/association_basics.html#self-joins

**Comment by neugierige** [comment]
this is better: http://stackoverflow.com/questions/6559164/rails-associations-has-many-through-but-same-model

**Comment by neugierige** [comment]
this is definitely outdated, but it may still be helpful: http://railscasts.com/episodes/163-self-referential-association

**Comment by payne-chris-r** [comment]
What's the update on this? I know you guys were looking at it together.

**Comment by neugierige** [comment]
I figured out the solution, but she's focusing on meeting CRUD requirements for now

**Comment by sujkid** [comment]
Lu helped in realizing this, apparently you cannot have two foreign keys that reference to the same table(!!!!!), 
I created the table, but I haven't implemented it yet.. I'll close the issue once it gets working for me.

**Comment by sujkid** [comment]
^ i haven't implement it = I haven't used the table

**Comment by sujkid** [comment]
In my original ERD, I had one table that had two foreign keys that refer to the same table. After hitting a few roadblocks trying to implement it, and receiving lots of help from Lu and Rachel, I realized that it is possible to do so, but not through migrations.
Lu pointed me in the right direction and I followed the documentation here - http://www.spacevatican.org/2008/5/6/creating-multiple-associations-with-the-same-table/
and successfully created the table. For anyone in future with the same issue, when you create the table through migrations, add the columns that are intended to be the foreign keys as regular columns, but add _id at the end of the names so rails will recognize them as foreign keys. And in the models, make changes as per the documentation above. The rest as Lu says, is "rails magic". The table should work as intended.
Having said that, I have decided to change my table structure to have only a single foreign key for now because of time constraints. That will change a particular feature of my app slightly, but I am going to continue with it for the time being. I will come back and implement the way my app was originally supposed to be, and explore the foreign keys and the "rails magic", and update this issue when I do that.
Thank you Lu and Rachel for sticking with me!
